receiver/dcgmreceiver/util_test.go

```diff
@@ -18,103 +18,118 @@
+func TestMetricStatsRateInt64(t *testing.T) {
+	testMetricStatsRate[int64](t, fieldValueInt64)
 }
 
-func TestRateIntegratorFloat64(t *testing.T) {
-	testRateIntegrator[float64](t)
+func TestMetricStatsRateFloat64(t *testing.T) {
+	testMetricStatsRate[float64](t, fieldValueFloat64)
 }
 
-func testCumulativeTracker[V int64 | float64](t *testing.T) {
-	origNowUnixMicro := nowUnixMicro
-	nowUnixMicro = func() int64 { return 10 }
-	defer func() { nowUnixMicro = origNowUnixMicro }()
+func testMetricStatsCumulative[V int64 | float64](t *testing.T, fv func(*testing.T, int64, V) dcgm.FieldValue_v2) {
+	stats := &metricStats{}
 
 	type P struct {
 		ts int64
-		v  V
+		v  int64
+	}
+	p := func(stats *metricStats) P {
+		if stats.lastFieldValue == nil {
+			return P{0, stats.cumulativeValue}
+		}
+		return P{stats.lastFieldValue.Ts, stats.cumulativeValue}
 	}
-	p := func(ts int64, v V) P { return P{ts, v} }
-
-	var ct cumulativeTracker[V]
 
-	ct.Reset()
-	require.Equal(t, P{0, 0}, p(ct.Baseline()))
-	require.Equal(t, P{10, 0}, p(ct.Value()))
+	require.Equal(t, int64(0), stats.initialCumulativeValue)
+	require.Equal(t, P{0, 0}, p(stats))
 	// Ensure first updates sets the baseline.
-	ct.Update(15, 50)
-	require.Equal(t, P{15, 50}, p(ct.Baseline()))
-	assert.Equal(t, P{15, 0}, p(ct.Value()))
+	stats.Update(fv(t, 15, 50))
+	require.Equal(t, int64(50), stats.initialCumulativeValue)
+	assert.Equal(t, P{15, 0}, p(stats))
 	// Ensure updates affect values, but not the baseline.
-	ct.Update(20, 80)
-	assert.Equal(t, P{15, 50}, p(ct.Baseline()))
-	assert.Equal(t, P{20, 30}, p(ct.Value()))
+	stats.Update(fv(t, 20, 80))
+	assert.Equal(t, int64(50), stats.initialCumulativeValue)
+	assert.Equal(t, P{20, 30}, p(stats))
 	// Ensure stale points are ignored.
-	ct.Update(18, 1e8)
-	assert.Equal(t, P{20, 30}, p(ct.Value()))
-	ct.Update(20, 1e8)
-	assert.Equal(t, P{20, 30}, p(ct.Value()))
+	stats.Update(fv(t, 18, 1e8))
+	assert.Equal(t, P{20, 30}, p(stats))
+	stats.Update(fv(t, 20, 1e8))
+	assert.Equal(t, P{20, 30}, p(stats))
 	// Ensure updates affect values.
-	ct.Update(25, 100)
-	assert.Equal(t, P{25, 50}, p(ct.Value()))
+	stats.Update(fv(t, 25, 100))
+	assert.Equal(t, P{25, 50}, p(stats))
 	// Ensure same inputs don't affect values.
-	ct.Update(30, 100)
-	assert.Equal(t, P{30, 50}, p(ct.Value()))
-
-	// Ensure the value and baseline are cleared on reset.
-	ct.Reset()
-	assert.Equal(t, P{0, 0}, p(ct.Baseline()))
-	assert.Equal(t, P{10, 0}, p(ct.Value()))
+	stats.Update(fv(t, 30, 100))
+	assert.Equal(t, P{30, 50}, p(stats))
 }
 
-func TestCumulativeTrackerInt64(t *testing.T) {
-	testCumulativeTracker[int64](t)
+func TestMetricStatsCumulativeInt64(t *testing.T) {
+	testMetricStatsCumulative[int64](t, fieldValueInt64)
 }
 
-func TestCumulativeTrackerFloat64(t *testing.T) {
-	testCumulativeTracker[float64](t)
+func TestMetricStatsCumulativeFloat64(t *testing.T) {
+	testMetricStatsCumulative[float64](t, fieldValueFloat64)
 }
 
 func TestDefaultMap(t *testing.T) {
```

Comment:
Also unused — let's remove?

Reply:
How is this unused? This is testing to make sure that we correctly calculate the cumulative value of a float64 metric.  It seems fine to me to keep that test, whether or not we're currently scraping a metric of that shape.

Reply:
I meant that the `defaultMap` functionality is unused, and thus we don't need `TestDefaultMap`.